### PR TITLE
KFedak - Task1

### PR DESCRIPTION
**Pull request by KateFedak**

### CODE REVIEW

File: Bootcamp/Task1/Program.cs [hunk 1]
```diff
@@ -0,0 +1,181 @@
+﻿using System;
+
+namespace Task1
+{
+    class Program
+    {
+        static void Main(string[] args)
```
Comment (olehkrut) [inline Bootcamp/Task1/Program.cs:7]: code style suggestion: you could separate each task with a new line, so it's easier to read

File: Bootcamp/Task1/Program.cs [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+
+
+        }
+
+        public static void NameOfAndInterpolation()
+        {
+            var number = 10;
+            var word = "dog";
+            Console.WriteLine($"Variable number and it`s value  {number}");
```
Comment (olehkrut) [inline Bootcamp/Task1/Program.cs:60]: probably you wanted to use `nameof(number)` right?

File: Bootcamp/Task1/Program.cs [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+
+        }
+        public static void FiveUpperChar(string str)
+        {
+
+            Console.WriteLine(str.Length < 5 ? $"String: {str} " : $"String:{str.Substring(0, 5).ToUpper() + str.Substring(5, str.Length - 5)}");
+        }
+        public static void Multiple3Or7(int number)
+        {
+            Console.WriteLine(number % 3 == 0 || number % 7 == 0 ? "true" : "false");
```
Comment (olehkrut) [inline Bootcamp/Task1/Program.cs:101]: there is a `Console.WriteLine(bool)` overload so you could omit `?` operator, unless you really wanted to use it)

File: Bootcamp/Task1/Program.cs [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+        }
+        public static void StartWithCSharp(string str)
+        {
+            bool first = false;
+            if (string.IsNullOrEmpty(str))
```
Comment (olehkrut) [inline Bootcamp/Task1/Program.cs:106]: there is a `StartsWith` method, easy as that

File: Bootcamp/Task1/Program.cs [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+
+        }
+        public static void MultiplicationTable(int number)
+        {
+
+            for (int i = 1; i <= number; i++)
+            {
+                for (int j = 1; j <= number; j++)
+                {
+                    if (j <= number - 1)
```
Comment (olehkrut) [inline Bootcamp/Task1/Program.cs:149]: hmmm, it seems like if and else clauses are the same, aren't they?

File: Bootcamp/Task1/Program.cs [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+            int count = 0;
+            int x = numbers[0];
+            for (int i = numbers[0]; i < numbers[1]; i++)
+            {
+                if (i % 8 == 0)
+                {
+                    count++;
+                    x = i;
+                }
+                else if (numbers[0] <= x && x <= numbers[1])
```
Comment (olehkrut) [inline Bootcamp/Task1/Program.cs:170]: it's better to check if the range is valid before doing the actual loop, just check whether leftNumber > rightNumber

Comment (olehkrut) [reply Bootcamp/Task1/Program.cs:170]: you can also increase i += 8 when i is divisible by 8 - small optimization

**Review by olehkrut — COMMENTED** [review]
Looks nice! 
One suggestion for future PRs: pls separate each method with a new line, it just looks better

File: Bootcamp/Task1/Program.cs [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+        }
+        public static void StartWithCSharp(string str)
+        {
+            bool first = false;
+            if (string.IsNullOrEmpty(str))
+            {
+                Console.WriteLine("The string is empty!");
+            }
+            var array = str.Split(new string[] { " " }, StringSplitOptions.None);
+            if (array[0].Substring(0, 2).ToLower() == "c#")
```
Comment (m96roman) [inline Bootcamp/Task1/Program.cs:111]: Can be just
```
bool first = str.Substring(0, 2).ToLower() == "c#";
Console.WriteLine(first);
```

File: Bootcamp/Task1/Program.cs [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+            Console.WriteLine(number % 3 == 0 || number % 7 == 0 ? "true" : "false");
+        }
+        public static void StartWithCSharp(string str)
+        {
+            bool first = false;
+            if (string.IsNullOrEmpty(str))
+            {
+                Console.WriteLine("The string is empty!");
+            }
+            var array = str.Split(new string[] { " " }, StringSplitOptions.None);
```
Comment (m96roman) [inline Bootcamp/Task1/Program.cs:110]: You don't need this split.

File: Bootcamp/Task1/Program.cs [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+            }
+            else
+            {
+                first = false;
+                Console.WriteLine($"{first}");
+            }
+        }
+        public static void CirclePerimeterAndArea(double radius)
+        {
+            double per_cir, area;
```
Comment (m96roman) [inline Bootcamp/Task1/Program.cs:124]: Don't do like this.  Each variable better to declare on separate line.
```

            double per_cir = 2 * Math.PI * radius;
            double area = Math.PI * radius * radius;
```

File: Bootcamp/Task1/Program.cs [hunk 10]
```diff
@@ -0,0 +1,181 @@
...
+            per_cir = 2 * Math.PI * radius;
+            area = Math.PI * radius * radius;
+            Console.WriteLine($"Perimeter of Circle {per_cir}");
+            Console.WriteLine($"Area of circle: {area}");
+        }
+        public static void SumAndAvg(int[] numbers)
+        {
+            int sum = 0;
+            double avg = 0;
```
Comment (m96roman) [inline Bootcamp/Task1/Program.cs:133]: No need to declare this variable here. 
You can declare it, where you use it.

File: Bootcamp/Task1/Program.cs (hunk 5 above)
Comment (m96roman) [reply Bootcamp/Task1/Program.cs:149]: You don't need if/else at all. Just Console.Write(.....)

**Review by m96roman — APPROVED** [review]

File: Bootcamp/Task1/Program.cs (hunk 2 above)
Comment (KateFedak) [reply Bootcamp/Task1/Program.cs]: fixed

File: Bootcamp/Task1/Program.cs (hunk 3 above)
Comment (KateFedak) [reply Bootcamp/Task1/Program.cs]: fixed

File: Bootcamp/Task1/Program.cs (hunk 4 above)
Comment (KateFedak) [reply Bootcamp/Task1/Program.cs:121]: fixed

File: Bootcamp/Task1/Program.cs (hunk 5 above)
Comment (KateFedak) [reply Bootcamp/Task1/Program.cs]: fixed

File: Bootcamp/Task1/Program.cs (hunk 6 above)
Comment (KateFedak) [reply Bootcamp/Task1/Program.cs]: fixed

File: Bootcamp/Task1/Program.cs (hunk 7 above)
Comment (KateFedak) [reply Bootcamp/Task1/Program.cs]: fixed

File: Bootcamp/Task1/Program.cs (hunk 8 above)
Comment (KateFedak) [reply Bootcamp/Task1/Program.cs]: fixed

File: Bootcamp/Task1/Program.cs (hunk 9 above)
Comment (KateFedak) [reply Bootcamp/Task1/Program.cs]: fixed

File: Bootcamp/Task1/Program.cs (hunk 10 above)
Comment (KateFedak) [reply Bootcamp/Task1/Program.cs]: fixed

**Pull request closed without merging**